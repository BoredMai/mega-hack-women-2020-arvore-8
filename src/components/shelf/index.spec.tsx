import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Shelf from '.';

describe('<Shelf>', () => {
  const items = [
    { icon: 'princess.png', label: 'Princesa', url: 'princesa' },
    {
      disabled: true,
      icon: 'wizard.png',
      label: 'Mago',
      url: 'mago',
    },
    {
      disabled: true,
      icon: 'unicorn.png',
      label: 'UnicÃ³rnio',
      url: 'unicornio',
    },
  ];
  const props = {
    items,
    onConfirm: jest.fn().mockName('onConfirm'),
  };
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Shelf {...props} />) as ShallowWrapper;
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('renders properly with custom label', () => {
    wrapper.setProps({ label: 'Continuar' });
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  describe('with single item', () => {
    beforeEach(() => {
      wrapper.setProps({ items: [items[0]] });
    });

    it('renders properly', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('calls prop on selection confirm', () => {
      wrapper.find('button').simulate('click');

      expect(props.onConfirm).toHaveBeenCalledWith(items[0]);
    });
  });

  describe('with item selection', () => {
    beforeEach(() => {
      wrapper.find('Item').at(0).simulate('select', items[0]);
    });

    it('selects item on item click', () => {
      expect(wrapper.find('button')).toHaveLength(1);
    });

    it('calls prop on selection confirm', () => {
      wrapper.find('button').simulate('click');

      expect(props.onConfirm).toHaveBeenCalledWith(items[0]);
    });
  });
});
