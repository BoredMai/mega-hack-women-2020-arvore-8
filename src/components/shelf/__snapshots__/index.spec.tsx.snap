// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shelf> renders properly 1`] = `
<React.Fragment>
  <div
    className="shelf"
  >
    <Item
      isSelected={false}
      item={
        Object {
          "icon": "princess.png",
          "label": "Princesa",
          "url": "princesa",
        }
      }
      onSelect={[Function]}
    />
    <Item
      isSelected={false}
      item={
        Object {
          "disabled": true,
          "icon": "wizard.png",
          "label": "Mago",
          "url": "mago",
        }
      }
      onSelect={[Function]}
    />
    <Item
      isSelected={false}
      item={
        Object {
          "disabled": true,
          "icon": "unicorn.png",
          "label": "Unicórnio",
          "url": "unicornio",
        }
      }
      onSelect={[Function]}
    />
  </div>
</React.Fragment>
`;

exports[`<Shelf> renders properly with custom label 1`] = `
<React.Fragment>
  <div
    className="shelf"
  >
    <Item
      isSelected={false}
      item={
        Object {
          "icon": "princess.png",
          "label": "Princesa",
          "url": "princesa",
        }
      }
      onSelect={[Function]}
    />
    <Item
      isSelected={false}
      item={
        Object {
          "disabled": true,
          "icon": "wizard.png",
          "label": "Mago",
          "url": "mago",
        }
      }
      onSelect={[Function]}
    />
    <Item
      isSelected={false}
      item={
        Object {
          "disabled": true,
          "icon": "unicorn.png",
          "label": "Unicórnio",
          "url": "unicornio",
        }
      }
      onSelect={[Function]}
    />
  </div>
</React.Fragment>
`;

exports[`<Shelf> with single item renders properly 1`] = `
<button
  onClick={[Function]}
>
  Avançar
</button>
`;
